lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,112 @@
+package handler
+
+import (
+	"context"
+	"log"
+
+	domainmodels "code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+)
+
+// Handler handles bets and event updates.
+type Handler struct {
+	betRepository BetRepository
+}
+
+// New creates and returns a new Handler.
+func New(betRepository BetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+// HandleBets handles bets.
+func (h *Handler) HandleBets(
+	ctx context.Context,
+	bets <-chan rabbitmqmodels.Bet,
+) <-chan rabbitmqmodels.BetCalculated {
+	resultingBets := make(chan rabbitmqmodels.BetCalculated)
+
+	go func() {
+		defer close(resultingBets)
+
+		for bet := range bets {
+			log.Println("Processing bet, betId:", bet.Id)
+
+			// Calculate the domain bet based on the incoming bet received.
+			domainBet := domainmodels.BetCalculated{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			_, exists, _ := h.betRepository.GetByID(ctx, domainBet.Id)
+			if !exists {
+				// Insert the domain bet into the repository.
+				err := h.betRepository.InsertBet(ctx, domainBet)
+				if err != nil {
+					log.Println("Failed to insert bet, error: ", err)
+					continue
+				}
+			}
+		}
+	}()
+
+	return resultingBets
```

Comment:
You will never send anything into this channel. That can be resolved in two ways, I'm fine with both:
- change method signature and don't return anything
- send all messages to `resultingBets`, but don't call producer from `engine.go`

Reply:
I've resolved this issue according to the first option.

Reply:
Seems like it's not resolved? I think you've messed it up and changed HandleEventUpdates signature.
Btw. you need to change interface definition in `engine/handler.go` because otherwise your struct won't implement the interface.
You also shouldn't even create `resultingBets` variable in this method implementation as you're not returning it.

Reply:
Oh yeah, I see now that it's messed up and that I've refactored the wrong function. The problem should be fixed now.